### DISCUSSION TITLE
let updating modpack via .mrpack in already created profile

### DISCUSSION THREAD

**Discussion by ghost** [modrinth/.github#292] [Feature requests]
I am not gonna make my modpack public, and morinth doesn't allow make it private without long inspection that doesnt allow make modpack of everything, so i have to use .mrpack, but it lack of "update" feature, so my homies have to create new profile each update of pack, download all data again and transit all setting files from previous profile. Inconvenient.

and i am not only who thinks so
https://support.modrinth.com/en/articles/8797522-sharing-modpacks
https://github.com/orgs/modrinth/discussions/266

**Comment by Nidoskull** [comment]
Seconding this; it's rather frustrating on the player side of this. Not only do I have to delete & recreate profiles, if I want to *keep* anything from it I'd need to manually move it over before re-creating the profile.
Terrible UX if you have a changing modlist *of any kind* for a server.

I'd see this implemented by checking the modlist of the existing instance and adding and updating mods as you go. If you have mods that are *not* in the new .mrpack, show a pop-up and let the user decide what to do with those particular mods (as they could be clientside mods intentionally added by the player, *or* mods that were fully removed from the pack for a reason.)
